akka-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala

```diff
@@ -10,10 +10,12 @@ import akka.parboiled2._
 import akka.shapeless._
 import akka.http.scaladsl.model._
 
+private[http] case class HeaderParserSettings(uriParsingMode: Uri.ParsingMode = Uri.ParsingMode.Relaxed)
```

Comment:
If we really need to wrap the `uriParsingMode`would it be better to cache the default instance in an `object`?